optuna/_convert_positional_args.py

```diff
@@ -5,18 +5,42 @@
 from functools import wraps
 from inspect import Parameter
 from inspect import signature
+import textwrap
 from typing import Any
 from typing import TYPE_CHECKING
 from typing import TypeVar
 import warnings
 
+from optuna._deprecated import _validate_two_version
+from optuna._experimental import _validate_version
+
 
 if TYPE_CHECKING:
     from typing_extensions import ParamSpec
 
     _P = ParamSpec("_P")
     _T = TypeVar("_T")
 
+_DEPRECATION_NOTE_TEMPLATE = """
+
+.. warning::
+    Using positional arguments is deprecated since v{d_ver}. Support for positional arguments
+    will be removed in the future. The removal is currently scheduled for v{r_ver}, but this
+    schedule is subject to change. Please use keyword arguments instead.
+    See https://github.com/optuna/optuna/releases/tag/v{d_ver}.
+"""
+
+_DEPRECATION_WARNING_TEMPLATE = (
+    "Positional arguments in {name} have been deprecated since v{d_ver}. "
```

Comment:
The warning message currently only shows the function name, but it would be more helpful to also display which positional arguments have been deprecated.